### ISSUE TITLE
How To / Add .done method for new Response();

### ISSUE THREAD

**Issue by leonbakadorov** [denoland/deno#23486]
Add a .done method to new Response() to ensure that the response was delivered to the client.

```
function handler(_req) {
    const res = new Response(null, { headers });

    if(res.done) {
        // ... Execute if the response is guaranteed to be delivered to the client.
    }

    return res; 
}
Deno.serve({ port: 80, handler });
```

**Comment by leonbakadorov** [comment]
> Why don't use `res.ok`?

The fact is that “res.ok” allows you to verify only the successful generation of the response, but not its guaranteed delivery to the client.

**Comment by leonbakadorov** [comment]
> We will not add non-standard methods to `Response`. There's a PR open that will provide other API that will provide this functionality: #23425

Yes. This is a very important functionality that ensures predictable and reliable system operation. It is important that this functionality be deterministic.

Tell me, in which version of Deno do you plan to release this?)

**Comment by mmastrac** [comment]
Fixed by https://github.com/denoland/deno/pull/23425 which will be released tomorrow.

**Comment by leonbakadorov** [comment]
> Fixed by #23425 which will be released tomorrow.

How to use it? How can you ensure that the Response is guaranteed to be received by the client?)
